azkaban-common/src/main/java/azkaban/server/SlaRequestUtils.java

```diff
@@ -26,28 +30,52 @@
 import java.util.List;
 import java.util.Map;
 import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
 import org.apache.log4j.Logger;
 
+
 public class SlaRequestUtils {
 
+  public static final String PARAM_SLA_EMAILS = "slaEmails";
+  public static final String PARAM_SLA_ALERTERS = "slaAlerters";
   public static final String SLA_STATUS_SUCCESS = "SUCCESS";
   private static final Logger logger = Logger.getLogger(SlaRequestUtils.class);
 
-  public static List<SlaOption> parseSlaOptions(final String flowName, final String emailStr,
-      final Map<String, String> settings) throws ServletException {
-    final List<String> slaEmails;
-    if (emailStr == null) {
-      slaEmails = Arrays.asList();
+  public static List<SlaOption> parseSlaOptions(final HttpServletRequest req, final String flowName,
```

Comment:
[non-blocker, improvement] I am curious to know the reason behind sending the servlet request object here. Does it make testing this SlaRequestUtils harder? The parseSlaOptions is doing quite a lot of work. Should we consider adding the reasoning or adding a todo for testing this?

Reply:
The parsing of SLA options is done in 2 different places, when SLAs are created and new executions are submitted. We moved common duplicated logic like capturing SLA email param into the `parseSlaOptions`. Based on that we might need to adjust the method name accordingly, you're right there.

I think tests in `HttpRequestUtilsTest` and `ExecutorServletTest` classes check this parsing logic indirectly but please let me know if something is missing and I'll add it in a follow up PR.

Reply:
Thanks Yeni for the clarification!